script.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  # my_each
+  def my_each
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    i = 0
+    while i < to_a.length
+      yield to_a[i], index(to_a[i])
+      i += 1
+    end
+    self
+  end
```

Comment:
- [ ] should have working ```my_each_with_index```  identical and returns the same thing as ruby's each_with_index. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

-  enumerable ```my_each_with_index``` returns an enumerator if no block is given
- It should work with ranges.
- It should work with hashes.